.vscode/settings.json

```diff
@@ -2,11 +2,17 @@
     "extensions.ignoreRecommendations": true,
     "C_Cpp.default.configurationProvider": "particle.particle-vscode-core",
     "files.associations": {
-        "*.ino": "cpp"
+        "*.ino": "cpp",
+        "cstdarg": "cpp"
     },
-    "particle.firmwareVersion": "5.3.0",
+    "particle.firmwareVersion": "4.1.0",
     "particle.targetPlatform": "tracker",
-    "particle.targetDevice": "",
+    "particle.targetDevice": "SmartFin-DevBoard-2",
     "cmake.configureOnOpen": false,
-    "C_Cpp.codeAnalysis.clangTidy.enabled": true
+    "C_Cpp.codeAnalysis.clangTidy.enabled": true,
+    "C_Cpp.codeAnalysis.clangTidy.checks.disabled": [
+        "clang-diagnostic-error",
+        "clang-diagnostic-error",
+        "clang-diagnostic-error"
+    ]
```

Comment:
Which diagnostic errors are we ignoring?

Reply:
Currently if not ignored vscode spits out a bunch of pretty useless diagnostic errors as the beggining of each file

Reply:
Here's some errors that are ignored:

`inet_hal_compat.h(53, 1): static assertion failed due to requirement 'sizeof(_HAL_IPAddress_t) == 17': HAL_IPAddress_size
static_assert.h(20, 49): expanded from macro 'PARTICLE_STATIC_ASSERT'
inet_hal_compat.h(53, 65): expression evaluates to '33 == 17'
static_assert.h(20, 63): expanded from macro 'PARTICLE_STATIC_ASSERT'
wlan_hal.h(242, 1): static assertion failed due to requirement 'sizeof(wlan_country_code_t) == 2': ISO 3166-1 alpha-2 country codes are two ASCII bytes
wlan_hal.h(242, 43): expression evaluates to '4 == 2'
spark_wiring_cellular_printable.h(94, 9): 'this_type_does_not_support_comparisons' is a private member of 'CellularData'
spark_wiring_cellular_printable.h(67, 10): declared private here
spark_wiring_cellular_printable.h(99, 9): 'this_type_does_not_support_comparisons' is a private member of 'CellularData'
spark_wiring_cellular_printable.h(67, 10): declared private here
spark_wiring_cellular_printable.h(146, 9): 'this_type_does_not_support_comparisons' is a private member of 'CellularBand'
spark_wiring_cellular_printable.h(111, 10): declared private here
spark_wiring_cellular_printable.h(151, 9): 'this_type_does_not_support_comparisons' is a private member of 'CellularBand'
spark_wiring_cellular_printable.h(111, 10): declared private here
new(89, 14): enumeration redeclared with different underlying type 'size_t' (aka 'unsigned int') (was 'unsigned long')
new(126, 26): 'operator new' takes type size_t ('unsigned long') as first parameter
new(128, 26): 'operator new[]' takes type size_t ('unsigned long') as first parameter
new(140, 26): 'operator new' takes type size_t ('unsigned long') as first parameter
new(142, 26): 'operator new[]' takes type size_t ('unsigned long') as first parameter
new(149, 57): no type named 'align_val_t' in namespace 'std'
new(151, 57): no type named 'align_val_t' in namespace 'std'
new(153, 34): no type named 'align_val_t' in namespace 'std'
new(155, 34): no type named 'align_val_t' in namespace 'std'
new(157, 59): no type named 'align_val_t' in namespace 'std'
new(159, 59): no type named 'align_val_t' in namespace 'std'
new(161, 36): no type named 'align_val_t' in namespace 'std'
new(163, 36): no type named 'align_val_t' in namespace 'std'`